src/daemon/daemon.cpp

```diff
@@ -3183,6 +3185,26 @@ grpc::Status mp::Daemon::shutdown_vm(VirtualMachine& vm, const std::chrono::mill
     return grpc::Status::OK;
 }
 
+grpc::Status mp::Daemon::switch_off_vm(VirtualMachine& vm)
+{
+    const auto& name = vm.vm_name;
+    const auto& state = vm.current_state();
+
+    using St = VirtualMachine::State;
+    const auto skip_states = {St::off, St::stopped};
```

Comment:
Shouldn't this skip `suspended` too?

Reply:
No, that makes total sense to me, I think that's the best. The code isn't doing that yet though, is it?

Reply:
> The code isn't doing that yet though, is it?

It does on the `hyperv` backend automatically, but not on the `qemu` backend.  I still have to confirm the behavior on the `virtualbox` and `libvirt` backends.

Reply:
I should add it's still on my TODO to implement the suspend image deletion on the backends that support suspend and it doesn't work yet.

Reply:
OK, makes sense, thanks!